pkg/cri/resource-manager/cache/utils.go

```diff
@@ -82,13 +86,46 @@ func estimateComputeResources(lnx *cri.LinuxContainerResources, cgroupParent str
 
 // SharesToMilliCPU converts CFS CPU shares to milliCPU.
 func SharesToMilliCPU(shares int64) int64 {
+	return roundMilliCPU(sharesToMilliCPU(shares))
+}
+
+// QuotaToMilliCPU converts CFS quota and period to milliCPU.
+func QuotaToMilliCPU(quota, period int64) int64 {
+	return quotaToMilliCPU(quota, period)
+}
+
+// sharesToMilliCPU converts CFS CPU shares to milliCPU.
+func sharesToMilliCPU(shares int64) int64 {
 	if shares == minShares {
 		return 0
 	}
-
 	return int64((float64(shares*milliCPUToCPU) / float64(sharesPerCPU)) + 0.5)
 }
 
+// quotaToMilliCPU converts CFS quota and period to milliCPU.
+func quotaToMilliCPU(quota, period int64) int64 {
+	if quota == 0 || period == 0 {
+		return 0
+	}
+	return int64(float64(quota*milliCPUToCPU)/float64(period) + 0.5)
+}
+
+// roundMilliCPU rounds values within milliCPUThreshold to milliCPUResolution.
+func roundMilliCPU(value int64) int64 {
+	diff := value % milliCPUResolution
+	if diff == 0 {
+		return value
+	}
+
+	if diff != 0 && diff < milliCPUThreshold {
```

Comment:
No need to check if `diff != 0` since it was made certain in the above `if` statement.

Reply:
Decided to remove the rounding code altogether for now. Instead, we now simply check that for full CPU's our estimates are accurate and otherwise the estimates are within our expected accuracy. Also, we switched to using the relevant constants (and also functions whenever possible) directly from kubelet as those are now exported.